cli/tests/test_checkout.rs

```diff
@@ -100,7 +100,80 @@ fn test_checkout_not_single_rev() {
     "###);
 }
 
+#[test]
+fn test_checkout_conflicting_branches() {
+    // Much of this test is borrowed from `test_git_fetch_conflicting_branches` in
+    // test_git_fetch.rs
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    add_git_remote(&test_env, &repo_path, "rem1");
+
+    // Create a rem1 branch locally
+    test_env.jj_cmd_success(&repo_path, &["new", "root()"]);
+    test_env.jj_cmd_success(&repo_path, &["branch", "create", "rem1"]);
+
+    test_env.jj_cmd_success(
+        &repo_path,
+        &["git", "fetch", "--remote", "rem1", "--branch", "*"],
+    );
```

Comment:
We don't need to involve a remote in order to create a conflicted branch. You can do it like this:

```
jj init --git
jj describe -m one
jj new -m two @-
jj branch create foo
jj --at-op=@- branch create foo -r 'description("one")'
```

That creates two commits and simulates two concurrent commands creating a branch called "foo" pointing at the two different commits